src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp

```diff
@@ -222,7 +222,7 @@ int Flasher::upgradeBMCFirmware(firmwareImage* bmc)
  * readback from flash and save in the file specified
  * only qspi_ps is supported currently
  */
-void Flasher::readBack(const std::string& output)
+int Flasher::readBack(const std::string& output)
```

Comment:
Unfortunately, the previous code reviews missed this method.  What needs to be done here is:
1. Remove the return statements.   
    - There is need need to have them in the switch's case body for they do nothing.
3. Remove the output prints (e.g., std::cout << "Error ...).
    - This causes a bigger issue for the code reports an error but there is no error return code.
4. When there is an error, throw an exception.
5. If needed, update the calling methods to this method to catch the exception.

Reply:
Noted